public/app/plugins/panel/geomap/editor/LayerDragDropList/LayerDragDropList.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { DragDropContext, Draggable, Droppable, DropResult } from 'react-beautiful-dnd';
+import { css, cx } from '@emotion/css';
+import { Icon, IconButton, stylesFactory } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+import { LayerName } from './LayerName';
+
+type LayerDragDropListProps = {
+  layers: any[];
+  onDragEnd: (result: DropResult) => void;
+  onSelect: (element: any) => any;
+  onDelete: (element: any) => any;
+  onDuplicate?: (element: any) => any;
+  selection?: Number[];
+  excludeBaseLayer?: boolean;
+  verifyLayerNameUniqueness?: (nameToCheck: string) => boolean;
+};
+
+export const LayerDragDropList = ({
+  layers,
+  onDragEnd,
+  onSelect,
+  onDelete,
+  onDuplicate,
+  selection,
+  excludeBaseLayer,
+  verifyLayerNameUniqueness,
+}: LayerDragDropListProps) => {
+  const style = styles(config.theme);
+
+  const getRowStyle = (isSelected: boolean) => {
+    return isSelected ? `${style.row} ${style.sel}` : style.row;
+  };
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+      <Droppable droppableId="droppable">
+        {(provided, snapshot) => (
+          <div {...provided.droppableProps} ref={provided.innerRef}>
+            {(() => {
+              // reverse order
+              const rows: any = [];
+              const lastLayerIndex = excludeBaseLayer ? 1 : 0;
+              for (let i = layers.length - 1; i >= lastLayerIndex; i--) {
+                const element = layers[i];
+                let uid = element.UID;
+                if (element.options.name) {
+                  uid = element.options.name;
+                }
+
+                const isGroup = element?.item?.id === 'group';
+
+                const isSelected = Boolean(selection?.find((id) => id === element.UID || id === i));
+                rows.push(
+                  <Draggable key={uid} draggableId={uid} index={rows.length}>
+                    {(provided, snapshot) => (
+                      <div
+                        className={getRowStyle(isSelected)}
+                        ref={provided.innerRef}
+                        {...provided.draggableProps}
+                        {...provided.dragHandleProps}
+                        onMouseDown={() => onSelect(element)}
+                      >
+                        <LayerName layer={element} verifyLayerNameUniqueness={verifyLayerNameUniqueness ?? undefined} />
+                        <div className={style.textWrapper}>&nbsp; {element.options.type}</div>
```

Comment:
```suggestion
                        <div className={style.textWrapper}>&nbsp; {props.getInfo(element)}</div>
```
Maybe a callback function to get the text?